src/app/meshes/views/MeshListView.vue

```diff
@@ -1,124 +1,123 @@
+          <template #title>
+            <h1>
+              <RouteTitle
+                :title="t('meshes.routes.items.title')"
+                :render="true"
               />
+            </h1>
+          </template>
+          <div class="stack">
+            <KCard>
+              <template #body>
+                <ErrorBlock
+                  v-if="error !== undefined"
+                  :error="error"
+                />
 
-              <AppCollection
-                v-else
-                class="mesh-collection"
-                data-testid="mesh-collection"
-                :headers="[
-                  { label: 'Name', key: 'name' },
-                  { label: 'Actions', key: 'actions', hideLabel: true },
-                ]"
-                :page-number="props.page"
-                :page-size="props.size"
-                :total="data?.total"
-                :items="data?.items"
-                :error="error"
-                :empty-state-message="t('common.emptyState.message', { type: 'Meshes' })"
-                :empty-state-cta-to="t('meshes.href.docs')"
-                :empty-state-cta-text="t('common.documentation')"
-                @change="route.update"
-              >
-                <template #name="{ row: item }">
-                  <RouterLink
-                    :to="{
-                      name: 'mesh-detail-view',
-                      params: {
-                        mesh: item.name,
-                      },
-                    }"
-                  >
-                    {{ item.name }}
-                  </RouterLink>
-                </template>
-                <template #actions="{ row: item }">
-                  <KDropdownMenu
-                    class="actions-dropdown"
-                    :kpop-attributes="{ placement: 'bottomEnd', popoverClasses: 'mt-5 more-actions-popover' }"
-                    width="150"
-                  >
-                    <template #default>
-                      <KButton
-                        class="non-visual-button"
-                        appearance="secondary"
-                        size="small"
-                      >
-                        <template #icon>
-                          <KIcon
-                            :color="KUI_COLOR_TEXT_NEUTRAL_STRONGER"
-                            icon="more"
-                            :size="KUI_ICON_SIZE_30"
-                          />
-                        </template>
-                      </KButton>
-                    </template>
+                <AppCollection
+                  v-else
+                  class="mesh-collection"
+                  data-testid="mesh-collection"
+                  :headers="[
+                    { label: 'Name', key: 'name' },
+                    { label: 'Actions', key: 'actions', hideLabel: true },
+                  ]"
+                  :page-number="parseInt(route.params.page)"
+                  :page-size="parseInt(route.params.size)"
```

Comment:
**Issue**: This will yield to `NaN` values in the case of parameter absence.

Reply:
The idea is these values should never be absent, if they are then its a bug in RouteView and should be fixed there.

Reply:
But we already know that they will always be absent initially. Easy enough to pass in sensible values from the start.

Reply:
Hmm maybe I'm not following or missing something here, the default values are here:

https://github.com/kumahq/kuma-gui/pull/1493/files#diff-0bd076fe74003e778a25db73b4e23416ba30f352d6e97fe4894acad30d018772R10-R13

```javascript
<RouteView
      v-if="me"
      v-slot="{ route, t }"
      name="mesh-list-view"
      :params="{
        page: 1,
        size: me.pageSize
      }"
>
```

`page` defaults to 1 and `size` defaults to whatever is return by `/me` (either a localStorage value or `50`). Thats the idea anyway, let me know if that isn't working for you.

Reply:
Any router navigation will go to a URL without query parameters unless we explicitly specify them (which we won’t) so there is always a window where those values can’t be set on the route level.

Reply:
> Any router navigation will go to a URL without query parameters

Yes, and the point of this PR is to redirect/replace that URL (without adding to history) with a URL with the query parameters. Can you explain where the window is when these values can't be set?

Reply:
> Can you explain where the window is when these values can't be set?

I did already:

> Any router navigation will go to a URL without query parameters unless we explicitly specify them (which we won’t) so there is always a window where those values can’t be set on the route level.

If you navigate to `/meshes`, for example, then that’s a case where that route won’t have these parameters.

Reply:
Right and then they are immediately set with the defaults? I put a console log in the routeParams computed function in RouteView and it does seem to me like they are always set for the `mesh-list-view` route? Are you not seeing the same?

Reply:
Okay, so when we use `route` from RouteView, as long as `props.params` has those defaults, that would work. How can we guarantee `props.params` has those values for list views? Looks like if that’s not set, those values will come through as `NaN`.

Also, this means we _can’t_ use vue-router’s `route` for this data the same way.

Reply:
> Okay, so when we use route from RouteView, as long as props.params has those defaults, that would work.

Cool ok I think you get this now. Glad you think this idea "has legs"/would work.

> How can we guarantee props.params has those values for list views? Looks like if that’s not set, those values will come through as NaN.

As of right now our tests should catch this, I'm pretty sure we never visit a page using `?queryParams=`. But if we think this PR is a good approach, then `route.params.page` wouldn't ever be set unless you gave it a default on the RouteView#params (even if a user decided to just add `?page=something` to the URL). If we do think this is a good approach then this PR (or a quickish follow up) will add the necessary default params to every route, and then add the work to not even set the `route.param` if a default isn't set. Then you would immediately see if you hadn't set the default by accident.